tensorflow/lite/micro/kernels/xtensa_hifimini/svdf.cc

```diff
@@ -91,168 +114,89 @@ void EvalIntegerSVDF(TfLiteContext* context, TfLiteNode* node,
   int16_t* const state_ptr = GetTensorData<int16_t>(activation_state_tensor);
 
   // Left shift the activation_state.
+  if(((int)state_ptr&0x3) == 0)
```

Comment:
avoid c-style cast + split out this logic with a comment for clarify.

Reply:
updated based on comment